src/celeritas/optical/OpticalCollector.hh

```diff
@@ -78,7 +77,7 @@ class OpticalCollector
 
   public:
     // Construct with core data and optical params
-    OpticalCollector(CoreParams const&, Input&&);
+    OpticalCollector(SPConstCore, Input&&);
```

Comment:
Why is this change needed? The optical collector shouldn't keep a copy of the core. It doesn't look like you're using it as a shared pointer.

Reply:
CoreParams is a shared_ptr in the Runner class. Passing the shared_ptr seemed more natural than the object's reference, but I can revert that.

Reply:
No, the runner holds a copy of it because it uses the core data. The optical collector merely *uses* the core params to construct its equivalent.